packages/core/src/app/payment/Payment/Payment.tsx

```diff
@@ -0,0 +1,453 @@
+            currentSelected.skipRedirectConfirmationAlert) {
+            return;
+        }
+
+        const message = language.translate('common.leave_warning');
+
+        event.returnValue = message;
+
+        return message;
+    }, [defaultMethod, selectedMethod, isSubmittingOrder]);
+
+    const handleSubmit = useCallback(async (values: PaymentFormValues) => {
+        const method = selectedMethod || defaultMethod;
+
+        analyticsTracker.clickPayButton({ shouldCreateAccount: values.shouldCreateAccount });
+
+        const customSubmit = method && submitFunctions[getUniquePaymentMethodId(method.id, method.gateway)];
+
+        if (customSubmit) {
+            return customSubmit(values);
+        }
+
+        try {
+            const state = await submitOrder(mapToOrderRequestBody(values, isPaymentDataRequired()));
+            const order = state.data.getOrder();
+
+            analyticsTracker.paymentComplete();
+            onSubmit(order?.orderId);
+        } catch (error) {
+            analyticsTracker.paymentRejected();
+
+            if (isErrorWithType(error) && error.type === 'payment_method_invalid') {
+                return loadPaymentMethods();
+            }
+
+            if (isCartChangedError(error)) {
+                return onCartChangedError(error);
+            }
+
+            onSubmitError(error as Error);
+        }
+    }, [defaultMethod, selectedMethod]);
+
+    const handleCloseModal = useCallback(async (_: Event, { error }: ErrorModalOnCloseProps) => {
+        if (!error) {
+            return;
+        }
+
+        const { type: errorType } = error as any;
+
+        if (errorType === 'provider_fatal_error' || errorType === 'order_could_not_be_finalized_error') {
+            window.location.replace(cartUrl || '/');
+        }
+
+        if (errorType === 'tax_provider_unavailable') {
+            window.location.reload();
+        }
+
+        if (errorType === 'cart_consistency') {
+            await loadCheckout();
+        }
+
+        if (isErrorWithType(error) && error.body) {
+            const { body, headers, status } = error;
+
+            if (body.type === 'provider_error' && headers.location) {
+                window.top?.location.assign(headers.location);
+            }
+
+            if (status === 429 || body.type === 'spam_protection_expired' || body.type === 'spam_protection_failed') {
+                setDidExceedSpamLimit(true);
+                await loadCheckout();
+            }
+        }
+
+        void clearError(error);
+    }, []);
+
+    const paymentContextValue = useMemo(() => ({
+        disableSubmit,
+        setSubmit,
+        setValidationSchema,
+        hidePaymentSubmitButton,
+    }), [disableSubmit, setSubmit, setValidationSchema, hidePaymentSubmitButton]);
+    const uniqueSelectedMethodId = defaultMethod && getUniquePaymentMethodId(
+        selectedMethod?.id ?? defaultMethod?.id,
+        selectedMethod?.gateway ?? defaultMethod?.gateway
+    );
```

Comment:
In the previous implementation, the default value of selected payment method was missed, hence validation schema was not set.
https://github.com/bigcommerce/checkout-js/blob/c031a523b7012d14aa1c8cf1bfaa50cba48e08c6/packages/core/src/app/payment/Payment.tsx#L182